packages/bigcommerce-payments-utils/src/bigcommerce-payments-fastlane-utils.ts

```diff
@@ -98,24 +98,31 @@ export default class BigCommercePaymentsFastlaneUtils {
      *
      * 'updateStorageSessionId' method is used to:
      * - set session id after user was authenticated (or unrecognised) to trigger authentication after page refresh
-     * - remove sessionId from browser storage if the customer canceled PayPal Fastlane Authentication
      *
      * Flow info:
      * If user unrecognised then the lookup method will be working but the OTP will not be shown
      * If user recognised and not canceled then the lookup method will be working and the OTP will be shown only if needed
      * If user cancels the OPT then OTP will not be triggered after page refresh
      *
      */
-    updateStorageSessionId(shouldBeRemoved: boolean, sessionId?: string): void {
-        if (shouldBeRemoved) {
-            this.browserStorage.removeItem('sessionId');
-        } else {
-            this.browserStorage.setItem('sessionId', sessionId);
-        }
+    updateStorageSessionId(sessionId: string): void {
+        const expires = new Date();
+        expires.setDate(expires.getDate() + 14); // 2 weeks expiry
+
+        const cookiesOption = {
+            expires,
+            secure: true,
+        };
+
+        CookieStorage.set('bc-fastlane-sessionId', sessionId, cookiesOption);
+    }
+
+    removeStorageSessionId(): void {
+        CookieStorage.remove('bc-fastlane-sessionId', { secure: true });
```

Comment:
Do we need to provide `{ secure: true }` when we removing cookie by name?

Reply:
Good catch 👍 
I just checked it once again. It is not necessary to provide secure param, when removing cookie. I will remove it